app/assets/javascripts/hyrax/collections/editor.es6

```diff
@@ -6,12 +6,17 @@ import tabifyForm from 'hyrax/tabbed_form'
 // Add search for thumbnail to the edit descriptions
 export default class {
   constructor(elem) {
-    let url =  window.location.pathname.replace('edit', 'files')
     let field = elem.find('#collection_thumbnail_id')
-    this.thumbnailSelect = new ThumbnailSelect(url, field)
+    this.thumbnailSelect = new ThumbnailSelect(this.url(), field)
     tabifyForm(elem.find('form.editor'))
 
     let participants = new Participants(elem.find('#participants'))
     participants.setup()
   }
+
+  url() {
+    let urlParts = window.location.pathname.split("/")
+    urlParts[urlParts.length - 1] = "files"
```

Comment:
This logic appears to change the behavior. Instead of subbing "edit" for "files" it always replaces the trailing slug with "files". What happens if the trailing slug is not "edit"?